firewall/firewall_configuration_provider.go

```diff
@@ -0,0 +1,106 @@
+package firewall
+
+import (
+	"errors"
+	"github.com/xwb1989/sqlparser"
+	"gopkg.in/yaml.v2"
+	"github.com/cossacklabs/acra/firewall/handlers"
+)
+
+const BlacklistConfigStr = "blacklist"
+const WhitelistConfigStr = "whitelist"
+
+var ErrQuerySyntaxError = errors.New("fail to parse specified query")
+var ErrStructureSyntaxError = errors.New("fail to parse specified structure")
+
+type FirewallConfig struct {
+	Handlers []struct {
+		Handler string
+		Queries []string
+		Tables  []string
+		Rules   []string
+	}
+}
+
+func (firewall *Firewall) SetFirewallConfiguration(configuration []byte) error {
+
+	err := updateFirewall(firewall, configuration)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateFirewall(firewall *Firewall, configuration []byte) error {
+
+	var firewallConfiguration FirewallConfig
+
+	err := yaml.Unmarshal(configuration, &firewallConfiguration)
+	if err != nil {
+		return err
+	}
+
+	//fmt.Println(firewallConfiguration)
+
+	var firewallCheckers []QueryHandlerInterface
+	for _, handlerConfiguration := range firewallConfiguration.Handlers {
+		switch handlerConfiguration.Handler{
+		case WhitelistConfigStr:
+			whitelistHandler := &handlers.WhitelistHandler{}
+
+			whitelistHandler.AddQueries(handlerConfiguration.Queries)
+			whitelistHandler.AddTables(handlerConfiguration.Tables)
+			whitelistHandler.AddRules(handlerConfiguration.Rules)
+
+			firewallCheckers = append(firewallCheckers, whitelistHandler)
```

Comment:
and then we can call `firewall.AddHandler(whitelistHandler)` without need to extra check at end of loop